src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1288,30 +1285,43 @@ private void CalculateIORedirection(out bool redirectOutput, out bool redirectEr
             }
         }
 
-        private bool ValidateExtension(string path)
+        // On Windows, check the extension list and see if we should try to execute this directly.
+        // Otherwise, use the platform library to check executability
+        private bool IsExecutable(string path)
         {
-            // Now check the extension and see if it's one of the ones in pathext
+#if UNIX
+
+            if (Platform.NonWindowsIsExecutable(this.Path))
```

Comment:
Why not just:

> return Platform.NonWindowsIsExecutable(this.Path);

Reply:
From PowerShell Committee conclusion we should fallback on Unix to read PATHEXT.

Reply:
Additional code would need to be added to support something like executing `foo` where `foo.py` exists and `.py` was added to `PATHEXT` on Unix since Windows takes care of this automatically.  That is a feature and not bug fix so certainly out of scope of this PR.  I'd also wait until there's customer request for such a feature.

Reply:
@SteveL-MSFT I fixed your request. Tracking issue #7755 .